docs/simnet_environment.mediawiki

```diff
@@ -0,0 +1,321 @@
+[[./assets/dcr_tmux_simnet_setup.sh|provided in the repository at docs/assets/dcr_tmux_simnet_setup.sh]].
+
+This script makes use of [https://github.com/tmux/tmux tmux], which is
+typically readily available on Unix and Linux platforms, to setup a
+self-contained environment with several terminal-based windows and panes with
+<code>dcrd</code> and <code>dcrwallet</code> already configured.
+
+===2.1 Preconfigured Environment Overview===
+
+The following provides an overview of how the environment that is automatically
+setup by the aforementioned script is configured:
+
+* Multiple windows, each with two panes
+** The upper pane is either a <code>dcrd</code> instance or a <code>dcrwallet</code> instance
+** The lower pane provides a shell already in the directory associated with the process running in the upper pane
+* Two <code>dcrd</code> instances running simnet (named <code>dcrd1</code> and <code>dcrd2</code>)
+** The second connects to the first via a persistent/permanent connection
+* Two <code>dcrwallet</code> instances also running simnet (named <code>wallet1</code> and <code>wallet2</code>)
+** Both wallets are configured with the same seed
+** The <code>wallet1</code> instance:
+*** Connects to the <code>dcrd1</code> instance via JSON-RPC
+*** Automatically purchases tickets via the integrated ticket buyer
+*** Automatically votes
+** The <code> wallet2</code> instance:
+*** Connects to the <code>dcrd2</code> instance via JSON-RPC
+*** Automatically votes
+* Helper scripts accessible from the lower pane in each window that communicate with the associated process in the upper pane
+* Scripts available in the directories of all instances:
+** <code>ctl</code> - Invokes <code>dcrctl</code> communicating to the associated instance process
+** <code>stopall</code> - Stops all of the processes and closes the <code>tmux</code> session
+* Additional scripts for the <code>dcrd1</code> and <code>dcrd2</code> instances:
+** <code>mine</code> - Mines a specified number of blocks with a delay between
+* Additional scripts for the <code>dcrd2</code> instance:
+** <code>reorg</code> - Forces a chain reorganization in the instance
+* Additional scripts for the <code>wallet1</code> and <code>wallet2</code> instances:
+** <code>tickets</code> - Manually purchases the specified number of tickets with a default of 1
+** <code>xfer</code> - Creates a transaction that transfers (sends) funds to an address owned by the instance
+
+===2.2. Using the Preconfigured Environment===
+
+In order to get started, first ensure that the <code>dcrd</code>,
+<code>dcrwallet</code>, and <code>dcrctl</code> binaries you want to test with
+are available in your system path:
+
+  $ dcrd -V
+  dcrd version x.y.z (Go version <version> <os>)
+  $ dcrwallet -V
+  dcrwallet version x.y.z (Go version <version> <os>)
+  $ dcrctl -V
+  dcrctl version x.y.z (Go version <version> <os>)
+
+Once that above has been verified, run the setup script:
+
```

Comment:
Add a command to make script executable ?

Reply:
Good suggestion.  Added.